collections_hierarchy/mixins.py

```diff
@@ -1,87 +1,98 @@
 class ComparableMixin(object):
     def __eq__(self, other):
-        pass
+        return self == other
 
     def __ne__(self, other):
         return not self == other
 
     def __lt__(self, other):
-        pass
+        return len(self) < len(other)
 
     def __gt__(self, other):
-        pass
+        return len(self) > len(other)
 
     def __le__(self, other):
-        pass
+        return len(self) <= len(other)
 
     def __ge__(self, other):
-        pass
+        return len(self) >= len(other)
 
 
 class SequenceMixin(object):
     def __iter__(self):
-        pass
+        return iter(self)
 
     def __next__(self):
-        pass
+        if isinstance(self.values, list):
+            if self.i > len(self.values) - 1:
+                raise StopIteration
+            else:
+                self.i += 1
+                return self.values[self.i]
+                
+        #if isistance(self.values, dict):
+                
+                 
+            
 
     next = __next__
 
     def __len__(self):
-        pass
+        return len(self.values)
 
     def __getitem__(self, key):
-        pass
+        return self.values[key]
 
     def __setitem__(self, key, value):
-        pass
+        self.values[key] = value
 
     def __delitem__(self, key):
-        pass
+        del self.values[key]
 
     def __contains__(self, item):
-        pass
+        return item in self
 
 
 class RepresentableMixin(object):
     def __repr__(self):
-        pass
+        return "%s" % type(object)
 
     def __str__(self):
-        pass
-
+        #return("{}".format(self)) -> recursion error
+        return "%s" % self
+        
 
 class ConstructibleMixin(object):
     DATA_ATTR_NAME = 'data'
 
     def __init__(self, initial=None):
-        pass
+        self.initial = initial
 
 
 class OperableMixin(object):
     def __add__(self, other):
-        pass
+        return self + other
```

Comment:
Infinite recursion since it will try to implement `add` will call itself. You could try something like:

    lowercase_list = self.values + other.values
    uppercase_list = self.__class__(lowercase_list)
    return uppercase_list

The idea is you first utilize the add method for python's built in lists. Then you can convert that to a List object and return it.